lib/src/utils.dart

```diff
@@ -247,3 +247,13 @@ Stream<ProcessSignal> getSignals() => Platform.isWindows
     ? ProcessSignal.sigint.watch()
     : StreamGroup
         .merge([ProcessSignal.sigterm.watch(), ProcessSignal.sigint.watch()]);
+
+/// Build a list like "A, B and 3 more".
+String buildSample<T>(List<T> items, String format(T item)) {
+  final fullCount = items.length <= 3 ? items.length : 2;
+  var sample = items.take(fullCount).map(format).join(', ');
+  if (items.length > fullCount) {
+    sample = '$sample and ${items.length - fullCount} more.';
```

Comment:
Is a string buffer more efficient?

Reply:
It is, but the overhead of this is so little (compared to the entire process of the analysis) that I lean more towards compact code over the most efficient code. Can change if you feel strongly about it.